services/vision/server_test.go

```diff
@@ -170,12 +171,14 @@ func TestServerCaptureAllFromCamera(t *testing.T) {
 		det1 := objectdetection.NewDetection(image.Rectangle{}, 0.5, "yes")
 		return viscapture.VisCapture{
 			Detections: []objectdetection.Detection{det1},
+			Extra:      extra,
 		}, nil
 	}
 	captAllResp, err := server.CaptureAllFromCamera(context.Background(), &captureRequest)
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, len(captAllResp.Detections), test.ShouldEqual, 1)
-	test.That(t, getDetectionsResp.Detections[0].GetClassName(), test.ShouldEqual, "yes")
+	test.That(t, captAllResp.Detections[0].GetClassName(), test.ShouldEqual, "yes")
+	test.That(t, captAllResp.Extra.AsMap(), test.ShouldResemble, map[string]interface{}{"foo": "GetDetections"})
```

Comment:
Also here, can you create a test to show that you don't get a fatal error if nil is passed in extra

Reply:
done